actions/maven/publish/action.yml

```
@@ -0,0 +1,117 @@
+    required: false
+  gpg-private-key:
+    description: "gpg-key-pass"
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout the project repository
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-project-checkout@v1.8.0-rc.1
+    - name: Set up Java for publishing to Maven Central Repository
+      uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2 # v3
+      env:
+        MAVEN_USERNAME: ${{ inputs.maven-username }}
+        MAVEN_PASSWORD: ${{ inputs.maven-password }}
+        GPG_KEY_PASS: ${{ inputs.gpg-key-pass }}
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        server-id: ossrh
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_PASSWORD
+        gpg-private-key: ${{ inputs.gpg-private-key }}
+        gpg-passphrase: GPG_KEY_PASS
+
+    - name: Download the slsa attestation
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-folder@v1.8.0-rc.1
+      with:
+        name: "${{ inputs.provenance-download-name }}"
+        path: slsa-attestations
+        sha256: "${{ inputs.provenance-download-sha256 }}"
+
+    - name: Download the target dir
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-folder@v1.8.0-rc.1
+      with:
+        name: target
+        path: ./
+        sha256: "${{ inputs.target-download-sha256 }}"
+
+    - name: Publish to the Maven Central Repository
+      shell: bash
+      env:
+        MAVEN_USERNAME: "${{ inputs.maven-username }}"
+        MAVEN_PASSWORD: "${{ inputs.maven-password }}"
+        GPG_KEY_PASS: "${{ inputs.gpg-key-pass }}"
+        SLSA_DIR: "${{ inputs.provenance-download-name }}"
+        PROVENANCE_FILES: "${{ inputs.provenance-download-name }}"
+      run: |
+        # Build and run custom plugin
+        cd plugin && mvn clean install && cd ..
+        # Re-indexing the secondary jar files for deploy
+        mvn javadoc:jar source:jar
+        # Retrieve project version
+        version=$(mvn org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate -Dexpression=project.version -q -DforceStdout)
+        ARTIFACTID=$(mvn org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate -Dexpression=project.artifactId -q -DforceStdout)
+        # Reset the environment variables add in the base provenance
+        FILES="slsa-attestations/${PROVENANCE_FILES}/${ARTIFACTID}-${version}.jar.intoto.build.slsa"
+        TYPES=slsa
+        CLASSIFIERS=jar.intoto.build
+        # Find all necessary built jar files and attach them to the environment variable deploy
+        # shellcheck disable=SC2044 # We don't spawn a new sub shell since we are appending to global env vars
+        for name in $(find ./ -name "$ARTIFACTID-$version-*.jar")
+        do
+          # shellcheck disable=SC1001 # shellcheck complains over \- but the line does what it should.
+          TARGET=$(echo "${name}" | rev | cut -d\- -f1 | rev)
```

You don't actually need the backslash do you?
```suggestion
          TARGET=$(echo "${name}" | rev | cut -d- -f1 | rev)
```

---

I believe I tested this a while back and we do need it. Is it a blocker to leave it, or can we defer it with an inline TODO?

---

It works fine for me (both bash and sh) and I'm pretty sure it works fine on GHA

---

You are right. Removed the `\`.